framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -191,6 +226,82 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		await setChainConnectorInfo(this._chainConnectorPluginDB, chainConnectorInfo);
 	}
 
+	public async createCCU(
+		sendingChainID: Buffer,
+		certificate: Certificate,
+		newCertificateThreshold: bigint,
+	): Promise<CrossChainUpdateTransactionParams | undefined> {
+		let activeBFTValidatorsUpdate: BFTValidator[];
+		let activeValidatorsUpdate: ActiveValidator[] = [];
+		let certificateThreshold = newCertificateThreshold;
+
+		const blockHeader = this._chainConnectorState.blockHeaders.find(
+			header => header.height === certificate.height,
+		)!;
+
+		const chainAccount = await this._mainchainAPIClient.invoke<ChainAccount>(
+			'interoperability_getChainAccount',
+			{ chainID: sendingChainID },
+		);
+
+		const certificateBytes = codec.encode(certificateSchema, certificate);
+
+		const certificateValidationResult = await this._validateCertificate(
+			certificateBytes,
```

Comment:
Why don't we encode the certificate in the `_validateCertificate` function? Also, can't the `codec.encode` function fail (for instance if the message does not follow the schema)?

Reply:
indeed, it can fail. Than it makes sense to me that we encode as part of the certificate validation, catch the encoding error and return it as a false result (I agree we should use more specific error types)

Reply:
@has5aan you can move encoding part inside `private async _validateCertificate(` and also return specific error message for different fail conditions for logging to atleast know the reason why certificate validation failed. https://github.com/LiskHQ/lisk-sdk/blob/d656b033dbebbaa61a5711148af9fc044c418ecf/framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts#L247